EKF/control.cpp

```diff
@@ -1003,26 +1003,37 @@ void Ekf::controlHeightFusion()
 
 	case VDIST_SENSOR_GPS:
 
-		// Determine if GPS should be used as the height source
-		if (do_range_aid && _range_sensor.isDataHealthy()) {
+		// Do switching between GPS and rangefinder if using range finder as
+		// a height source when close to ground and moving slowly
+		if (!_control_status_prev.flags.rng_hgt && do_range_aid && _range_sensor.isDataHealthy()) {
 			setControlRangeHeight();
-			fuse_height = true;
 
 			// we have just switched to using range finder, calculate height sensor offset such that current
 			// measurement matches our current height estimate
 			if (_control_status_prev.flags.rng_hgt != _control_status.flags.rng_hgt) {
```

Comment:
This check can be removed because we only enter that part of the code if `!_control_status_prev.flags.rng_hgt` and we set `_control_status.flags.rng_hgt` just above (in `setControlRangeHeight();`)

Reply:
Will fix